applications/SystemIdentificationApplication/custom_utilities/smooth_clamper.cpp

```diff
@@ -39,16 +40,42 @@ SmoothClamper<TContainerType>::SmoothClamper(
     KRATOS_CATCH("");
 }
 
+template<class TContainerType>
+double SmoothClamper<TContainerType>::ProjectForward(const double X) const
+{
+    const double x_tilde = std::clamp((X - mMin) / mDelta, 0.0, 1.0);
+    return mMin + x_tilde * x_tilde * (3.0 - 2.0 * x_tilde) * mDelta;
+}
+
+template<class TContainerType>
+double SmoothClamper<TContainerType>::CalculateForwardProjectionGradient(const double X) const
+{
+    const double x_tilde = std::clamp((X - mMin) / mDelta, 0.0, 1.0);
+    return 6 * x_tilde - 6 * x_tilde * x_tilde;
+}
+
+template<class TContainerType>
+double SmoothClamper<TContainerType>::ProjectBackward(const double Y) const
+{
+    double x_tilde;
+    if (Y < mMin) {
+        x_tilde = 0;
+    } else if (Y > mMax) {
+        x_tilde = 1.0;
+    } else {
+        const double y = (Y - mMin) / mDelta;
+        x_tilde = 0.5 - std::sin(std::asin(1.0 - 2.0 * y) / 3.0);
```

Comment:
This is for SIMP or for SI?

Reply:
y review PRs while you're on holiday at the best place on earth?

Reply:
this is for SI only, SIMP does not have this problem

Reply:
@matekelemen I was waiting in a line

Reply:
It is almost linear. As far as I see, we cannot have continuous gradients in a linear clamping function, so unfortunately we have to live with this. This is onlz used in SIApp. Sigmoidal is another variation of this in a way which we use for SIMP and thickness optimization